src/NServiceBus.Core/Impersonation/Windows/WindowsIdentityEnricher.cs

```diff
@@ -19,11 +18,7 @@ public void MutateOutgoing(LogicalMessage logicalMessage, TransportMessage trans
                 return;
             }
             var windowsIdentity = WindowsIdentity.GetCurrent();
-            if (windowsIdentity != null)
```

Comment:
Unfortunately @danielmarbach, it can be null, @SimonCropp and I have ran into this situation before.
BTW, that link is for 4.6.2 CLR, so things could have been updated, but I am 100% sure in previous versions we ran into this.
There is even an issue that a user raised. (I'll try to find it)